TypeScript support

jimmythecoder opened the pull request:
Included typings from definitely typed repo and updated to support new options. This is required to support TypeScript, the definitely typed repo is becoming out of date and no longer recommended way of including typings.

leongersen commented:
This is really great, thanks! I have some comments; I'll add them as a review.

leongersen commented on index.d.ts:
```diff
@@ -0,0 +1,259 @@
+// Type definitions for nouislider v10.1.0
```
Could you remove these comments? You can add the author information to `readme.md`.

leongersen replied:
Could you format this file using prettier?

leongersen commented on index.d.ts:
```diff
@@ -0,0 +1,259 @@
+		/**
+         * The start option sets the number of handles and their start positions, relative to range.
+         */
+		start: number | number[] | number[][];
```
these can be `string`, too.

leongersen commented on index.d.ts:
```diff
@@ -0,0 +1,259 @@
+         * format for step mode
+         * see noUiSlider format
+         */
+		format?: Object | ((...args: any[]) => any);
```
We should probably add a `Formatter` interface

jimmythecoder commented:
Thanks! I'll try get these changes done today and update. Awesome!

jimmythecoder commented:
@leongersen I have made those changes you mentioned, added Formatter interface, added string as a start type,  formatted using prettier also merged in wNumb typings? Not actually required, but what are your thoughts to keep the interfaces together? 

Thanks.

leongersen commented:
I think the wNumb typings should be in that repository. It meets the `Formatter` interface, too, so that should work.

leongersen commented:
Other than that, I think it might be helpful to replace the documentation comments with links to the relevant documentation, so it doesn't have to be maintained twice. There are some descriptions that don't match the properties that they are placed above. I can do that, if you'd like.

leongersen commented:
I'm going to close this PR as it has gone stale. Porting noUiSlider to Typescript is a project I want to work on, as soon as I get the time.
